src/data_access/FileUserDataAccessObject.java

```diff
@@ -93,29 +91,39 @@ public void save() throws IOException {
                 writer.write(line);
                 writer.newLine();
             }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
         }
     }
 
     /**
-     * Creates a user and saves them to the database
-     * @param username represents the username of the new user
-     * @param password represents the password of the new user
+     * Saves user to database
+     * @param user represents the user to save
      */
-    public void createUser(String username, String password) {
-        User user = userFactory.create(username, password);
-        accounts.put(username, user);
-        try {
-            save();
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
+    @Override
+    public void saveUser(User user) {
+        accounts.put(user.getUsername(), user);
+        save();
+    }
+
+    /**
+     * Saves user preferences to database
+     * @param username represents the username of the user to save to
+     * @param preference represents the new preference
+     */
+    @Override
+    public void saveUserPreference(String username, UserPreference preference) {
```

Comment:
You've dropped the s in some of the preferences methods and variables. Try to keep them all consistent for readability. For the methods and classes in my use case, I left the s in. We should prob try to keep them the same

Reply:
Like literally just add the s for every method, class, variable, anything that uses the word preference. I think it makes more sense to keep everything as "preferences."